openedx/core/djangoapps/credit/tests/test_api.py

```diff
@@ -514,6 +528,34 @@ def test_set_credit_requirement_status(self, mode):
     @ddt.data(
         *CourseMode.CREDIT_ELIGIBLE_MODES
     )
+    def test_set_credit_requirement_status_satisfied_to_failed(self, mode):
+
+        """
+        Test that if credit requirment status is set to `satisfied`, it
+        can not not be changed to `failed`
+        """
+        credit_course = self.add_credit_course()
+
+        # Enroll user and verify enrollment.
+        self.enroll(self.user, self.course_key, mode)
+        self.assertTrue(CourseEnrollment.is_enrolled(self.user, self.course_key))
+        self.assertTrue(CourseEnrollment.enrollment_mode_for_user(self.user, self.course_key), (mode, True))
+
+        # set requirements for course to earn credit
+        course_requirements = self._set_credit_course_requirements()
+
+        api.set_credit_requirement_status(self.user, self.course_key, "grade", "grade", status="satisfied")
```

Comment:
I think you don't need to pass the `status="satisfied"` here, because by default status is `satisfied`. 

[set_credit_requirement_status(user, course_key, req_namespace, req_name, status="satisfied", reason=None):](https://github.com/edx/edx-platform/blob/1224e341def0be460f9590853582f287d0382077/openedx/core/djangoapps/credit/api/eligibility.py#L202-L202)